recipes/libsigcpp/2.x.x/conanfile.py

```diff
@@ -0,0 +1,80 @@
+import os
+
+from conans import ConanFile, Meson, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class LibSigCppConan(ConanFile):
+    name = "libsigcpp"
+    homepage = "https://github.com/libsigcplusplus/libsigcplusplus"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "LGPL-3.0"
+    description = (
+        "libsigc++ implements a typesafe callback system for standard C++."
+    )
+    topics = ("conan", "libsigcpp", "callback")
+    generators = "pkg_config"
+    settings = "os", "compiler", "arch", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {"shared": False, "fPIC": True}
+
+    _meson = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if (
+            self.settings.compiler == "Visual Studio"
+            and "MT" in str(self.settings.compiler.runtime)
+            and self.options.shared
+        ):
+            raise ConanInvalidConfiguration(
+                "Visual Studio and Runtime MT is not supported for shared library."
+            )
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("libsigc++-{}".format(self.version), self._source_subfolder)
+
+    def _configure_build(self):
+        if self._meson:
+            return self._meson
+        self._meson = Meson(self)
+        self._meson.configure(
+            source_folder=self._source_subfolder,
+            build_folder=self._build_subfolder,
+        )
+        return self._meson
+
+    def build(self):
+        meson = self._configure_build()
+        meson.build()
+
+    def package(self):
+        self.copy("COPYING", dst="licenses", src=self._source_subfolder)
+        meson = self._configure_build()
+        meson.install()
+
+        tools.rmdir(os.path.join(self.package_folder, "lib/pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.includedirs = [
+            "include/sigc++-2.0",
+            "lib/sigc++-2.0/include",
```

Comment:
wouldn't it be better to move this folder in include folder like in 3.x.x recipe?

Reply:
I figured the authors would have a good reason to do so, although it is unknown to me too.

The packagers of my distro (Arch Linux) did not move it either:

```$ pacman -Ql libsigc++                                                                                                                                                                                                                                                                                                                         
libsigc++ /usr/
libsigc++ /usr/include/
libsigc++ /usr/include/sigc++-2.0/
libsigc++ /usr/include/sigc++-2.0/sigc++/
libsigc++ /usr/include/sigc++-2.0/sigc++/adaptors/
libsigc++ /usr/include/sigc++-2.0/sigc++/adaptors/adaptor_trait.h
libsigc++ /usr/include/sigc++-2.0/sigc++/adaptors/adaptors.h
libsigc++ /usr/include/sigc++-2.0/sigc++/adaptors/bind.h
libsigc++ /usr/include/sigc++-2.0/sigc++/adaptors/bind_return.h
libsigc++ /usr/include/sigc++-2.0/sigc++/adaptors/bound_argument.h
libsigc++ /usr/include/sigc++-2.0/sigc++/adaptors/compose.h
libsigc++ /usr/include/sigc++-2.0/sigc++/adaptors/deduce_result_type.h
libsigc++ /usr/include/sigc++-2.0/sigc++/adaptors/exception_catch.h
libsigc++ /usr/include/sigc++-2.0/sigc++/adaptors/hide.h
libsigc++ /usr/include/sigc++-2.0/sigc++/adaptors/lambda/
libsigc++ /usr/include/sigc++-2.0/sigc++/adaptors/lambda/base.h
libsigc++ /usr/include/sigc++-2.0/sigc++/adaptors/lambda/select.h
libsigc++ /usr/include/sigc++-2.0/sigc++/adaptors/retype.h
libsigc++ /usr/include/sigc++-2.0/sigc++/adaptors/retype_return.h
libsigc++ /usr/include/sigc++-2.0/sigc++/adaptors/track_obj.h
libsigc++ /usr/include/sigc++-2.0/sigc++/bind.h
libsigc++ /usr/include/sigc++-2.0/sigc++/bind_return.h
libsigc++ /usr/include/sigc++-2.0/sigc++/connection.h
libsigc++ /usr/include/sigc++-2.0/sigc++/functors/
libsigc++ /usr/include/sigc++-2.0/sigc++/functors/functor_trait.h
libsigc++ /usr/include/sigc++-2.0/sigc++/functors/functors.h
libsigc++ /usr/include/sigc++-2.0/sigc++/functors/mem_fun.h
libsigc++ /usr/include/sigc++-2.0/sigc++/functors/ptr_fun.h
libsigc++ /usr/include/sigc++-2.0/sigc++/functors/slot.h
libsigc++ /usr/include/sigc++-2.0/sigc++/functors/slot_base.h
libsigc++ /usr/include/sigc++-2.0/sigc++/limit_reference.h
libsigc++ /usr/include/sigc++-2.0/sigc++/reference_wrapper.h
libsigc++ /usr/include/sigc++-2.0/sigc++/retype_return.h
libsigc++ /usr/include/sigc++-2.0/sigc++/sigc++.h
libsigc++ /usr/include/sigc++-2.0/sigc++/signal.h
libsigc++ /usr/include/sigc++-2.0/sigc++/signal_base.h
libsigc++ /usr/include/sigc++-2.0/sigc++/slot.h
libsigc++ /usr/include/sigc++-2.0/sigc++/trackable.h
libsigc++ /usr/include/sigc++-2.0/sigc++/type_traits.h
libsigc++ /usr/include/sigc++-2.0/sigc++/visit_each.h
libsigc++ /usr/lib/
libsigc++ /usr/lib/libsigc-2.0.so
libsigc++ /usr/lib/libsigc-2.0.so.0
libsigc++ /usr/lib/libsigc-2.0.so.0.0.0
libsigc++ /usr/lib/pkgconfig/
libsigc++ /usr/lib/pkgconfig/sigc++-2.0.pc
libsigc++ /usr/lib/sigc++-2.0/
libsigc++ /usr/lib/sigc++-2.0/include/
libsigc++ /usr/lib/sigc++-2.0/include/sigc++config.h```

To me, it seems best to leave the package as untouched as possible.